Workflow/Tests/ApplyContextTests.swift

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Testing
+
+@testable import Workflow
+
+@MainActor
+struct ApplyContextTests {
+    @Test
+    func concreteApplyContextInvalidatedAfterUse() async throws {
+        var escapedContext: ApplyContext<EscapingContextWorkflow>?
+        let onApply = { (context: ApplyContext<EscapingContextWorkflow>) in
+            #expect(context[workflowValue: \.property] == 42)
```

Comment:
These are the first `#expect`-based tests in Workflow, I believe. Just to confirm, did you check that failures are caught in CI?

Reply:
thanks for the callout. yes, i did a 'canary' run [yesterday](https://github.com/square/workflow-swift/pull/350) with a test that should have failed CI, and it seemed to behave as expected